commerce-test/src/testIntegration/java/com/liferay/commerce/shipment/test/CommerceShipmentTest.java

```diff
@@ -0,0 +1,1021 @@
+			commerceOrder.getGroupId(), commerceOrder.getCommerceOrderId(),
+			commerceInventoryWarehouse.getCommerceInventoryWarehouseId());
+
+		List<CommerceOrderItem> commerceOrderItems =
+			_commerceOrderItemLocalService.getCommerceOrderItems(
+				commerceOrder.getCommerceOrderId(), QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS);
+
+		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
+			List<CommerceShipmentItem> commerceShipmentItems =
+				_commerceShipmentItemLocalService.getCommerceShipmentItems(
+					commerceOrderItem.getCommerceOrderItemId());
+
+			Assert.assertEquals(
+				commerceShipmentItems.toString(), 1,
+				commerceShipmentItems.size());
+		}
+	}
+
+	@Test(expected = CommerceShipmentInactiveWarehouseException.class)
+	public void testCreateShippingFromInactiveWarehouse() throws Exception {
+		frutillaRule.scenario(
+			"Attach a shipment to a item that is stocked in an inactive " +
+				"warehouse"
+		).given(
+			"An order with an order item associated to an inactive warehouse"
+		).when(
+			"I attach the shipment to the order and its items"
+		).then(
+			"An exception shall be raised"
+		);
+
+		CPInstance cpInstance = _createCPInstance();
+
+		CommerceChannel commerceChannel = _createCommerceChannel();
+
+		BigDecimal value = BigDecimal.valueOf(RandomTestUtil.nextDouble());
+
+		CommerceOrder commerceOrder =
+			CommerceTestUtil.createCommerceOrderForShipping(
+				_company.getGroupId(), _user.getUserId(),
+				_commerceCurrency.getCommerceCurrencyId(),
+				commerceChannel.getSiteGroupId(), value);
+
+		_commerceOrders.add(commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			commerceOrder.getCommerceCurrency(), commerceChannel, null, null,
+			null, commerceOrder);
+
+		CommerceInventoryWarehouse commerceInventoryWarehouse =
+			_createCommerceInventoryWarehouse(
+				commerceChannel.getCommerceChannelId(), false);
+
+		int quantity = 10;
+
+		CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+			_user.getUserId(), commerceInventoryWarehouse, cpInstance.getSku(),
+			quantity);
+
+		int orderedQuantity = 1;
+
+		CommerceOrderItem commerceOrderItem =
+			CommerceTestUtil.addCommerceOrderItem(
+				commerceOrder.getCommerceOrderId(),
+				cpInstance.getCPInstanceId(), orderedQuantity, commerceContext);
+
+		CommerceShipment commerceShipment =
+			CommerceShipmentTestUtil.createEmptyOrderShipment(
+				commerceOrder.getGroupId(), commerceOrder.getCommerceOrderId());
+
+		_commerceShipmentItemLocalService.addCommerceShipmentItem(
+			commerceShipment.getCommerceShipmentId(),
+			commerceOrderItem.getCommerceOrderItemId(),
+			commerceInventoryWarehouse.getCommerceInventoryWarehouseId(),
+			orderedQuantity,
+			ServiceContextTestUtil.getServiceContext(
+				commerceOrder.getGroupId()));
+	}
+
+	@Test(expected = NoSuchOrderException.class)
+	public void testCreateShippingWithNoOrder() throws Exception {
+		frutillaRule.scenario(
+			"Attach a shipment to a non existing order"
```

Comment:
should really be `nonexistent` rather than `non existing`. "non existing" is something you'd hear spoken, but it's not really correct written.